proposals/22-04-localize-command.md

```diff
@@ -0,0 +1,331 @@
+resources:
+  - ./localized-files/github.com/kubernetes-sigs/kustomize/examples/multibases
+  - ./localized-files/github.com/kubernetes-sigs/kustomize/examples/helloWorld/configMap.yaml
+```
+
+Now, I upload `localized-overlay` from my local setup to my company’s internal package management site. I change the
+commands in my CI/CD pipeline to pull `localized-overlay` before running `kustomize build localized-overlay`, and the
+command executes successfully!
+
+### Risks and Mitigations
+
+One could argue that while uploading the localized `newDir` to a repository, a user could accidentally leak Secrets that
+were originally remote in a more private repo. This event is not very likely, as the servers that the user intends to
+consume the localized kustomizations often only have access to internal, private networks and repos. Nonetheless, users
+should have enough context in the case of Secrets to make the right decision. Users should have a basic understanding of
+the files that their `target` kustomizations reference and of the files that they plan to upload to repos. Secret
+configurations are also not too difficult to identify.
+
+Exec binaries that KRM functions reference are a different story. `kustomize localize` downloads remote exec binaries
+that, if malicious, are capable of almost anything during subsequent `kustomize build` calls. The command mitigates this
+risk by leaving these downloaded exec binaries without executable permissions and warning the user, as mentioned in
+**Proposal**. `kustomize build` can only run the exec binary after the user deems the binary safe and changes its
+permissions.
+
+Still another risk may be that if a user's kustomization tree is large, `kustomize localize` may be copying files from
+unexpected locations. The command mitigates this risk with the `scope` flag. If not set, `kustomize localize` only
+copies files in `target`. Otherwise, the user specifies `scope`and understands that `kustomize localize` only copies
+files in `scope`. The qualification that the command can only localize `target`s that follow load restrictions helps
+mitigate this risk as well.
+
+### Dependencies
+
+N/A
+
+### Scalability
+
+Large kustomization trees slows the performance of `kustomize localize`. These trees can have large local subtrees, have
+large remote subtrees, be deeply nested, or be wide, with each overlay referencing multiple bases. Regardless of the
+cause, large kustomization trees inevitably takes longer to copy and download. Parts of the kustomize code are not
+thread-safe, which precludes parallel execution.
+
+On a separate note, a chain of remote kustomization directories in which the current kustomization file references the
+next remote kustomization directories could create a `newDir` with deeply nested `localized-files`. This directory
+structure would impede users’ navigation of `newDir`. However, this scenario should be unlikely as most kustomizations
+only consist of a few layers.
```

Comment:
Nit: It's likely to happen, but not what we're optimizing for. Especially because the intent isn't for users to be modifying the localized files and this need to navigate them.